### PR TITLE
docs: Updates README for beta release

### LINKED ISSUE #19: Populate README appropriately
Fill in the **Getting Started Section** similar to other Ion Hash Implementations. 

1. Add badges 
1. Example on how to import and use the library as a client 
1. Small example on how to hash Ion

### PR DESCRIPTION
**Pull request by therapon**
Issue #, if available: Closes #19 

Description of changes:

Adds basic info in the README file including sample usage of the library. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,36 +1,121 @@
...
+
+```Go
+
+// Create a hasher provider, using MD5
+hasherProvider := ionhash.NewCryptoHasherProvider("MD5")
+
+// Create an Ion reader over the input [1,2,3]
+ionReader := ion.NewReaderString("[1,2,3]")
+
+// Create a hash Reader
```
Comment (byronlin-bq) [inline README.md:31]: reader

File: README.md [hunk 2]
```diff
@@ -1,36 +1,121 @@
...
+You can start using ion-hash-go by simply importing it, e.g.,
+
+```Go
+import (
+	ionhash "github.com/amzn/ion-hash-go"
+)
+```
+
+## Generating a hash while reading
+
```
Comment (byronlin-bq) [inline README.md:22]: nit: blank line

File: README.md [hunk 3]
```diff
@@ -1,36 +1,121 @@
...
+```Go
+
+// Create a hasher provider, using MD5
+hasherProvider := ionhash.NewCryptoHasherProvider("MD5")
+
+// Create an Ion reader over the input [1,2,3]
+ionReader := ion.NewReaderString("[1,2,3]")
+
+// Create a hash Reader
+hashReader, err := ionhash.NewHashReader( ionReader, hasherProvider)
```
Comment (byronlin-bq) [inline README.md:32]:
```suggestion
hashReader, err := ionhash.NewHashReader(ionReader, hasherProvider)
```

File: README.md [hunk 4]
```diff
@@ -1,36 +1,121 @@
...
+
+```Go
+
+// Create a hasher provider, using MD5
+hasherProvider := ionhash.NewCryptoHasherProvider("MD5")
+
+// Create an Ion writer
+ionWriter := ion.NewTextWriter(new(bytes.Buffer))
+
+// Create a hash Writer
```
Comment (byronlin-bq) [inline README.md:63]:
```suggestion
// Create a hash writer
```

File: README.md [hunk 5]
```diff
@@ -1,36 +1,121 @@
...
+hashWriter, err := ionhash.NewHashWriter(ionWriter, hasherProvider)
+if err != nil {
+	panic(err)
+}
+
+// Write the list [1,2,3]
+
```
Comment (byronlin-bq) [inline README.md:70]: nit: remove blank line

File: README.md [hunk 6]
```diff
@@ -1,36 +1,121 @@
...
 [![license](https://img.shields.io/hexpm/l/plug.svg)](https://github.com/amzn/ion-hash-go/blob/master/LICENSE)
-[![docs](https://img.shields.io/badge/docs-api-green.svg?style=flat-square)](https://amzn.github.io/ion-hash-go/api)
+[![docs](https://img.shields.io/badge/docs-api-green.svg?style=flat-square)](https://pkg.go.dev/github.com/amzn/ion-hash-go?tab=doc)
 
 ## Getting Started
 
-TBD
+You can start using ion-hash-go by simply importing it, e.g.,
+
+```Go
```
Comment (byronlin-bq) [inline README.md:15]:
```Go

import (
	ionhash "github.com/amzn/ion-hash-go"
)

```

File: README.md [hunk 7]
```diff
@@ -1,36 +1,122 @@
-:warning: **This package is considered experimental, under active early development, and the API is subject to change.** :warning:
+** This package is consider beta. While the API is relatively stable it is still subject to change. **
```
Comment (tgregg) [inline README.md:1]: consider -> considered

**Pull request merged**